### PR TITLE
Avoid StaleElementReferenceError in OLM e2e tests

### PR DESCRIPTION
**Pull request by spadgett**
https://jira.coreos.com/browse/CONSOLE-876

### CODE REVIEW

File: frontend/integration-tests/views/olm-catalog.view.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+import { browser, $, $$, by, element, ExpectedConditions as until } from 'protractor';
 
 import * as crudView from './crud.view';
 
-export const entryRows = $$('.co-resource-list__item');
-export const entryRowFor = (name: string) => entryRows.filter((row) => row.$('.co-clusterserviceversion-logo__name__clusterserviceversion').getText()
-  .then(text => text === name || text.startsWith(name))).first();
+const rowSelector = '.co-resource-list__item';
+export const entryRows = $$(rowSelector);
+export const entryRowFor = (name: string) => element(by.cssContainingText(rowSelector, name));
```
Comment (spadgett) [inline frontend/integration-tests/views/olm-catalog.view.ts]: This is less precise, but it seems unlikely we'd have false matches. It should avoid the `StaleElementReferenceError`. I verified it correctly fails if the item is missing:

```
Failed Specs:

1) Installing a service from a Catalog Source : displays Catalog Source with expected available packages
   NoSuchElementError: No element found using locator: by.cssContainingText(".co-resource-list__item", "BOGUS")
```

File: frontend/integration-tests/tests/olm/descriptors.scenario.ts [hunk 2]
```diff
@@ -167,6 +167,7 @@ describe('Using OLM descriptor components', () => {
   });
 
   it('displays list containing custom resource', async() => {
+    await crudView.resourceRowsPresent();
```
Comment (spadgett) [inline frontend/integration-tests/tests/olm/descriptors.scenario.ts]: I needed this check to run the tests locally. It appears the loading indicator can disappear briefly while discovery is in flight, which causes the `rowForName` check to run before the content is loaded. I'm not sure why I see in locally, but not in Jenkins.

**Comment by spadgett** [comment]
Well, this got past the stale element error, but failed at

```
1) Interacting with the etcd OCS : creates etcd Operator `Deployment`
   Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
```

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
Ugh, looks like the tests hit a stale element error on `isDiplayed()` in the last run.

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by alecmerdler** [comment]
jenkins rebuild

**Pull request merged**